### PR TITLE
Add Feast Serving histogram metrics

### PR DESCRIPTION
**Pull request by terryyylim**
Signed-off-by: Terence <terencelimxp@gmail.com>

**What this PR does / why we need it**:
This PR adds histogram metrics for Feast serving, which would allow us to determine:
- amount of entity rows per request (buckets)
- amount of features per request (buckets)
- amount of feature tables per request (buckets)

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Users can gather entity row, features and feature table requested metrics in Feast Serving.
```

### CODE REVIEW

File: serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java [hunk 1]
```diff
@@ -289,6 +292,29 @@ private static boolean checkOutsideMaxAge(
...
+    Metrics.requestEntityCount.labels(project).observe(Double.valueOf(entityRows.size()));
+    Metrics.requestFeatureCount.labels(project).observe(Double.valueOf(featureReferences.size()));
+
+    Set<String> featureTableRefs =
+        featureReferences.stream()
+            .map(featureReference -> getFeatureTableStringRef(project, featureReference))
+            .collect(Collectors.toSet());
+    Metrics.requestFeatureTableCount
+        .labels(project)
+        .observe(Double.valueOf(featureTableRefs.size()));
```
Comment (zeitos) [inline serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java:315]: nit: looks like you use featureTableRefs just for counting distinct strings, consider doing
```
long countDistinct = featureReferences.stream().map(...)
                        .distinct()
                        .count();
....
        .observe(Double.valueOf(countDistinct));

```

File: serving/src/main/java/feast/serving/util/Metrics.java [hunk 2]
```diff
@@ -29,12 +29,31 @@
           .labelNames("method")
           .register();
 
-  public static final Counter requestCount =
-      Counter.build()
+  public static final Histogram requestEntityCount =
+      Histogram.build()
+          .buckets(1, 2, 5, 10)
```
Comment (pyalex) [inline serving/src/main/java/feast/serving/util/Metrics.java:34]: oh, there could be much more :)

Let's do `1, 2, 5, 10, 20, 50, 100, 200`

File: serving/src/main/java/feast/serving/util/Metrics.java [hunk 3]
```diff
@@ -29,12 +29,31 @@
...
+          .name("request_entity_count")
+          .subsystem("feast_serving")
+          .help("Number of entity rows per request")
+          .labelNames("project")
+          .register();
+
+  public static final Histogram requestFeatureCount =
+      Histogram.build()
+          .buckets(1, 2, 5, 10, 15, 20)
```
Comment (pyalex) [inline serving/src/main/java/feast/serving/util/Metrics.java:43]: 1, 2, 5, 10, 15, 20, 30, 50

File: serving/src/main/java/feast/serving/util/Metrics.java [hunk 4]
```diff
@@ -29,12 +29,31 @@
...
           .subsystem("feast_serving")
-          .help("number of feature rows requested")
-          .labelNames("project", "feature_name")
+          .help("Number of feature rows per request")
+          .labelNames("project")
+          .register();
+
+  public static final Histogram requestFeatureTableCount =
+      Histogram.build()
+          .buckets(1, 2, 5, 10)
```
Comment (pyalex) [inline serving/src/main/java/feast/serving/util/Metrics.java:52]: 1, 2, 5, 10, 20

**Review by pyalex — APPROVED** [review]

**Pull request merged**